app/js/enhancedDate.js

```diff
@@ -1,5 +1,129 @@
 /* 
  * add your module definition for enhancedDate here
  * assign the module to a global variable
- * /
+*/
 
+var enhancedDate = (function enhancedDate() {
+  'use strict';
+
+  //Private data
+  var dateData = {
+    weekday : ['Sunday', 'Monday', 'Tuesday', 'Wednesday',
+                 'Thursday', 'Friday', 'Saturday'],
+    month : ['January', 'February', 'March', 'April', 'May',
+               'June', 'July', 'August', 'September', 'October',
+               'November', 'December']
+  };
+
+  //Private function
+  function isDateSet(){
+    if(!Date.parse(dateData.date)) {
+      dateData.date = new Date();
+    }
+  }
+
+  //Public API
+  var publicAPI = {
+    /* setDate: Stores the passed-in date. Takes milliseconds after epoch or a 
+    date object. Do some type-checking to make sure you have a valid type. If 
+    no value is passed, default to now.*/
+    setDate : function setDate(inDate) {
+      if(Date.parse(inDate)){
+        dateData.date = inDate;
```

Comment:
This works, but keep in mind that by using the passed-in Date instance, you're getting a reference to an object that can be modified outside your module and your internally stored reference will be modified at the same time. Better to just create a new Date instance for your module: date = new Date(inDate);